paddle/fluid/operators/math/softmax_impl.h

```diff
@@ -38,16 +40,37 @@ template <typename DeviceContext, typename T, bool is_test, typename Enable>
 void SoftmaxFunctor<DeviceContext, T, is_test, Enable>::operator()(
     const DeviceContext& context, const int axis_dim,
     const framework::Tensor* X, framework::Tensor* Y) {
-  auto logits = EigenMatrix<T>::From(*X);
-  auto softmax = EigenMatrix<T>::From(*Y);
-
+  auto in_dims = X->dims();
   const int kBatchDim = 0;
   const int kClassDim = 1;
 
-  const int batch_size = logits.dimension(kBatchDim);
-  const int num_classes = logits.dimension(kClassDim);
+  const int num_classes = in_dims[kClassDim];
+  const int batch_size = in_dims[kBatchDim];
   const int num_remain = num_classes / axis_dim;
 
+  if (num_remain == 1 && platform::MayIUse(platform::avx)) {
```

Comment:
Could you please why num_remain has to be equal 1 ? I understand that efficient softmax code can be used when axis_dim == rank -1  eg. we noramlize after final dimension.  Why not has condition: axis_dim + 1 = in.dims.size()

Reply:
It's refer to [here](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/jit/refer/refer.h#L444)

Reply:
ok